### PR TITLE
NFC: Updating README.txt 

### PR DESCRIPTION
**Pull request by blnicho**
## Summary/Motivation:
Our project description [on PyPI](https://pypi.org/project/Pyomo/) is out of date and still points to the old trac site. This description is automatically generated from our `README.txt` file and cannot be edited on PyPI outside of cutting a release. This PR updates `README.txt` so that future releases will have an up-to-date description. I've also made a note in the release process documentation to take a look at this file when finalizing the release.

## Changes proposed in this PR:
- Update README.txt to point to readthedocs for documentation and Github for issue tracking

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by ghackebeil** [comment]
Can we merge README.md and README.txt as part of this. I don't think there is any reason to have two separate README files. PyPI is smart enough to find and use anything that looks like `README.{txt,md,rst}`

**Comment by blnicho** [comment]
@ghackebeil I think that's a great idea. I'll make that change.

File: README.md [hunk 1]
```diff
@@ -15,7 +15,21 @@
...
+ -  Linear programming
+ -  Quadratic programming
+ -  Nonlinear programming
+ -  Mixed-integer linear programming
+ -  Mixed-integer quadratic programming
+ -  Mixed-integer nonlinear programming
+ -  Mixed-integer stochastic programming
+ -  Generalized disjunctive programming
+ -  Differential algebraic equations
+ -  Bilevel programming
```
Comment (jsiirola) [inline README.md]: While we are at it, we should remove Bilevel, as that package is currently deprecated.

**Comment by blnicho** [comment]
This is a NFC so no need to wait for tests to pass before merging.

**Pull request merged**